test/e2e/resize_nodes.go

```diff
@@ -259,27 +270,49 @@ func resizeRC(c clientset.Interface, ns, name string, replicas int32) error {
 	return err
 }
 
-func getMaster(c clientset.Interface) string {
-	master := ""
+// getMaster populates the externalIP, internalIP and hostname fields of the master.
+// If any of these is unavailable, it is set to "".
+func getMaster(c clientset.Interface) Address {
+	master := Address{}
+
+	// Populate the internal IP.
+	eps, err := c.Core().Endpoints(api.NamespaceDefault).Get("kubernetes")
+	if err != nil {
+		framework.Failf("Failed to get kubernetes endpoints: %v", err)
+	}
+	if len(eps.Subsets) != 1 || len(eps.Subsets[0].Addresses) != 1 {
+		framework.Failf("There are more than 1 endpoints for kubernetes service: %+v", eps)
+	}
+	master.internalIP = eps.Subsets[0].Addresses[0].IP
+
+	// Populate the external IP/hostname.
+	url, err := url.Parse(framework.TestContext.Host)
+	if err != nil {
+		framework.Failf("Failed to parse hostname: %v", err)
+	}
+	if net.ParseIP(url.Host) != nil {
+		// TODO: Check that it is external IP (not having a reserved IP address as per RFC1918).
+		master.externalIP = url.Host
```

Comment:
I guess, if you assume we want to drop all traffic to the ip, what does the rule do? does it match on port?